File: informatika_ismeretek/kozep/2020_may/java/SudokuGUI.java
import java.awt.event.*;
import javax.swing.*;

public class SudokuGUI {

    public static void main(String[] args) {
        var content = new JPanel(null);
        var dimensionField = newTextField("4", 220, 15, 40, false);

        content.add(newLabel("Új feladvány mérete:", 15, 15));
        content.add(newButton("-", 150, 15, 50, e -> handleMinusDimensionButtonPress(dimensionField)));
        content.add(newButton("+", 280, 15, 50, e -> handlePlusDimensionButtonPress(dimensionField)));
        content.add(dimensionField);

        var userTextInput = newTextField(null, 15, 90, 480, true);
        var lengthLabel = newLabel("Hossz: 0", 15, 120);

        content.add(newLabel("Kezdőállapot:", 15, 60));
        content.add(userTextInput);
        content.add(lengthLabel);
        content.add(newButton("Ellenőrzés", 395, 125, 100, e -> handleCheckButtonPress(dimensionField, userTextInput)));

        userTextInput.addCaretListener(e -> lengthLabel.setText("Hossz: " + userTextInput.getText().length()));

        var frame = new JFrame("Sudoku-ellenőrző");
        frame.setSize(540, 210);
        frame.setLocationRelativeTo(null);
        frame.setResizable(false);
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setContentPane(content);
        frame.setVisible(true);
    }


    public static void handleMinusDimensionButtonPress(JTextField dimensionField) {
        var oldValue = Integer.parseInt(dimensionField.getText());
        var newValue = Math.max(oldValue - 1, 4);

        dimensionField.setText(Integer.toString(newValue));
    }

    public static void handlePlusDimensionButtonPress(JTextField dimensionField) {
        var oldValue = Integer.parseInt(dimensionField.getText());
        var newValue = Math.min(oldValue + 1, 9);

        dimensionField.setText(Integer.toString(newValue));
    }

    public static void handleCheckButtonPress(JTextField dimensionField, JTextField userTextInput) {
        var expectedDimension = Integer.parseInt(dimensionField.getText());
        var userInputTextLength = userTextInput.getText().length();
        var expectedLength = expectedDimension * expectedDimension;

        if(userInputTextLength == expectedLength) {
            JOptionPane.showMessageDialog(null, "A feladvány megfelelő hosszúságú!");
        }else{
            var smaller = Math.min(expectedLength, userInputTextLength);
            var larger = Math.max(expectedLength, userInputTextLength);
            var argumentPart = larger == expectedLength ? "rövid: kell még " : "hosszú: törlendő ";

            JOptionPane.showMessageDialog(null, "A feladvány + " + argumentPart + (larger - smaller) + "számjegy!");
        }
    }



    public static JLabel newLabel(String text, int x, int y) {
        var label = new JLabel(text);
        label.setBounds(x, y, text.length() * 7, 30);
        return label;
    }

    public static JButton newButton(String text, int x, int y, int width, ActionListener listener) {
        var butt = new JButton(text);
        butt.setBounds(x, y, width, 30);
        butt.addActionListener(listener);
        return butt;
    }

    public static JTextField newTextField(String text, int x, int y, int width, boolean editable) {
        var field = new JTextField(text);
        field.setBounds(x, y, width, 30);
        field.setEditable(editable);
        return field;
    }
}

File: informatika_ismeretek/kozep/2020_may/c#/SudokuGUI.cs
using System;
using System.Drawing;
using System.Windows.Forms;

var form = new Form{ Text = "Sudoku-ellenőrző", ClientSize = new Size(540, 210) };
var dimensionBox = new TextBox{ Location = new Point(220, 15), Size = new Size(40, 30), Text = "4", ReadOnly = true };
var minusDimensionButton = new Button{ Location = new Point(150, 15), Size = new Size(50, 30), Text = "-" };
var plusDimensionButton = new Button{ Location = new Point(280, 15), Size = new Size(50, 30), Text = "+" };

minusDimensionButton.Click += (s, e) => HandleMinusDimensionButtonPress(dimensionBox);
plusDimensionButton.Click += (s, e) => HandlePlusDimensionButtonPress(dimensionBox);

form.Controls.Add(new Label{ Location = new Point(15, 15), Size = new Size(150, 30), Text = "Új feladvány mérete:" });
form.Controls.Add(minusDimensionButton);
form.Controls.Add(dimensionBox);
form.Controls.Add(plusDimensionButton);

var userTextInput = new TextBox{ Location = new Point(15, 90), Size = new Size(480, 30) };
var lengthLabel = new Label{ Location = new Point(15, 120), Size = new Size(150, 30), Text = "Hossz: 0" };
var checkButton = new Button{ Location = new Point(395, 125), Size = new Size(100, 30), Text = "Ellenőrzés" };

form.Controls.Add(new Label{ Location = new Point(15, 60), Size = new Size(100, 30), Text = "Kezdőállapot" });
form.Controls.Add(userTextInput);
form.Controls.Add(lengthLabel);
form.Controls.Add(checkButton);

userTextInput.TextChanged += (s, e) => lengthLabel.Text = "Hossz: " + userTextInput.Text.Length;
checkButton.Click += (s, e) => HandleCheckButtonPress(dimensionBox, userTextInput);

Application.Run(form);


void HandleMinusDimensionButtonPress(TextBox dimensionField) {
    var oldValue = int.Parse(dimensionField.Text);
    var newValue = Math.Max(oldValue - 1, 4);

    dimensionField.Text = newValue.ToString();
}

void HandlePlusDimensionButtonPress(TextBox dimensionField) {
    var oldValue = int.Parse(dimensionField.Text);
    var newValue = Math.Min(oldValue + 1, 9);

    dimensionField.Text = newValue.ToString();
}

void HandleCheckButtonPress(TextBox dimensionBox, TextBox userTextInput) {
    var expectedDimension = int.Parse(dimensionBox.Text);
    var userInputTextLength = userTextInput.Text.Length;
    var expectedLength = expectedDimension * expectedDimension;

    if(userInputTextLength == expectedLength) {
        MessageBox.Show("A feladvány megfelelő hosszúságú!");
    }else{
        var smaller = Math.Min(expectedLength, userInputTextLength);
        var larger = Math.Max(expectedLength, userInputTextLength);
        var argumentPart = larger == expectedLength ? "rövid: kell még" : "hosszú: törlendő";

        MessageBox.Show($"A feladvány {argumentPart} {larger - smaller} számjegy!");
    }
}